uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js

```diff
@@ -169,6 +175,24 @@ define([
         : leaveRequests;
     }
 
+    /**
+     * Finds indexed leave request by a leave request given
+     *
+     * @param  {LeaveRequestInstance} leaveRequest
+     * @return {LeaveRequestInstance} indexed leave request
+     */
+    function findIndexedLeaveRequest (leaveRequest) {
+      var indexedLeaveRequest;
+
+      _.forEach(leaveRequests[leaveRequest.contact_id], function (day) {
+        indexedLeaveRequest = _.find(day, function (leaveRequestObj) {
+          return +leaveRequestObj.id === +leaveRequest.id;
+        }) || indexedLeaveRequest;
+      });
+
+      return indexedLeaveRequest;
+    }
```

Comment:
In case this is necessary, why not

```js
var indexedLeaveRequest;

// By using `find` here the loop will stop as soon as an indexedLeaveRequest is found:
_.find(leaveRequests[leaveRequest.contact_id], function (day) {
  indexedLeaveRequest = _.find(day, function (leaveRequestObj) {
    return +leaveRequestObj.id === +leaveRequest.id;
  });

  return indexedLeaveRequest;
});

return indexedLeaveRequest;
```

Reply:
Thanks a lot. Done.